sdfg_utils.js

```diff
@@ -196,19 +196,27 @@ function traverse_sdfg_scopes(sdfg, func, post_subscope_func=null) {
     scopes_recursive(sdfg, sdfg.nodes());
 }
 
+// contains already visited edges to speed up computation of memlet trees
+var memlet_tree_edges_visited = [];
+
 /**
- * Returns a partial memlet tree from a given edge, from the root node 
+ * Returns a partial memlet tree from a given edge, from the given node 
  * through all children (without siblings). Calling this function with
  * the root edge returns the entire memlet tree.
  **/
-function memlet_tree(graph, edge, root_only = false) {
```

Comment:
I would say it's better to keep the original memlet_tree function and add a new one (we might use it in the future).

Reply:
OK, I fixed this